contracts/proxy/ImplementationAuthority.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.6.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract ImplementationAuthority is Ownable {
```

Comment:
would put `is IImplementationAuthority` as well and create a separate `IImplepmentationAuthority` contract